contracts/libs/FixedPointInt256.sol

```diff
@@ -0,0 +1,63 @@
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ */
+pragma solidity 0.6.10;
+
+import "../packages/oz/SignedSafeMath.sol";
+
+/**
+ *
+ */
+library FixedPointInt256 {
+    using SignedSafeMath for int256;
+
+    int256 private constant SCALING_FACTOR = 1e18;
+
+    /**
+     * @notice convert an unsigned integer to signed integer
+     * @param a uint to convert into a signed integer.
+     * @return the converted signed integer.
+     */
+    function fromUint(uint256 a) internal pure returns (int256) {
+        require(a < uint256(-1), "FixedPointInt256: out of int range");
+
+        return int256(a);
+    }
+
+    /**
+     * @notice convert a signed integer to unsigned integer
+     * @param a int to convert into an unsigned integer.
+     * @return the converted unsigned integer.
+     */
+    function fromInt(int256 a) internal pure returns (int256) {
+        if (a < 0) {
+            uint256(-a);
+        } else {
+            uint256(a);
+        }
+    }
+
+    function add(int256 a, int256 b) internal pure returns (int256) {
+        return a.add(b);
+    }
+
+    function sub(int256 a, int256 b) internal pure returns (int256) {
+        return a.sub(b);
+    }
+
+    function mul(int256 a, int256 b) internal pure returns (int256) {
+        return (a.mul(b)).div(SCALING_FACTOR);
+    }
+
+    function div(int256 a, int256 b) internal pure returns (int256) {
+        return (a.div(b)).mul(SCALING_FACTOR);
```

Comment:
maybe you want to round these? See maker's math library for deets

Reply:
cool